x-pack/plugins/alerting/server/authorization/alerting_authorization.ts

```diff
@@ -388,30 +417,36 @@ export class AlertingAuthorization {
       return {
         username,
         hasAllRequested,
-        authorizedRuleTypes: hasAllRequested
-          ? // has access to all features
-            this.augmentWithAuthorizedConsumers(ruleTypes, await this.allPossibleConsumers)
-          : // only has some of the required privileges
-            privileges.kibana.reduce((authorizedRuleTypes, { authorized, privilege }) => {
-              if (authorized && privilegeToRuleType.has(privilege)) {
-                const [ruleType, feature, hasPrivileges, isAuthorizedAtProducerLevel] =
-                  privilegeToRuleType.get(privilege)!;
-                ruleType.authorizedConsumers[feature] = mergeHasPrivileges(
-                  hasPrivileges,
-                  ruleType.authorizedConsumers[feature]
-                );
-
-                if (isAuthorizedAtProducerLevel) {
-                  // granting privileges under the producer automatically authorized the Rules Management UI as well
-                  ruleType.authorizedConsumers[ALERTS_FEATURE_ID] = mergeHasPrivileges(
-                    hasPrivileges,
-                    ruleType.authorizedConsumers[ALERTS_FEATURE_ID]
-                  );
+        authorizedRuleTypes:
+          hasAllRequested && featuresIds === undefined
+            ? // has access to all features
+              this.augmentWithAuthorizedConsumers(
+                new Set(ruleTypesAuthorized.values()),
+                await this.allPossibleConsumers
+              )
+            : // only has some of the required privileges
+              privileges.kibana.reduce((authorizedRuleTypes, { authorized, privilege }) => {
+                if (authorized && privilegeToRuleType.has(privilege)) {
+                  const [ruleType, feature, hasPrivileges, isAuthorizedAtProducerLevel] =
+                    privilegeToRuleType.get(privilege)!;
+                  if ((featuresIds && featuresIds.has(feature)) || featuresIds === undefined) {
```

Comment:
would it simplify this code if we default `featuresIds` to an empty set? so we don't have to constantly do these checks

Reply:
Yes I can just use fIds and we should be good here